csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -31,6 +31,7 @@ public partial class Schema : IRecordType
         private readonly List<Field> _fieldsList;
 
         public ILookup<string, Field> FieldsLookup { get; }
+        private readonly ILookup<string, int> _fieldsIndexLookup;
```

Comment:
Can we also expose this in a way that lets users get all the indexes for a particular field name?